test/ant/worker/generate.xsl

```diff
@@ -66,31 +70,44 @@
 		<xsl:variable as="xs:boolean" name="enable-coverage"
 			select="processing-instruction(xspec-test) = 'enable-coverage'" />
 
+		<xsl:variable as="xs:boolean" name="xquery-3-1-required"
+			select="x:description/@xquery-version = '3.1'" />
+
 		<xsl:variable as="xs:string?" name="skip">
 			<xsl:choose>
 				<xsl:when test="
 						$enable-coverage
 						and not($XSLT-SUPPORTS-COVERAGE)">
-					<xsl:value-of>
-						<xsl:text>Skipping </xsl:text>
-						<xsl:value-of select="x:filename-and-extension($xspec-file-uri)" />
-						<xsl:text>: Requires XSLT processor to support coverage</xsl:text>
-					</xsl:value-of>
+					<xsl:text>Requires XSLT processor to support coverage</xsl:text>
 				</xsl:when>
 
 				<xsl:when
 					test="
 						(processing-instruction(xspec-test) = 'require-xslt-to-support-schema')
 						and not($XSLT-SUPPORTS-SCHEMA)">
-					<xsl:value-of>
-						<xsl:text>Skipping </xsl:text>
-						<xsl:value-of select="x:filename-and-extension($xspec-file-uri)" />
-						<xsl:text>: Requires schema-aware XSLT processor</xsl:text>
-					</xsl:value-of>
+					<xsl:text>Requires schema-aware XSLT processor</xsl:text>
+				</xsl:when>
+
+				<xsl:when
+					test="
+						$xquery-3-1-required
+						and not($XQUERY-SUPPORTS-3-1-DEFAULT or $XQUERY-SUPPORTS-3-1-QVERSION)">
+					<xsl:text>Requires XQuery 3.1 processor</xsl:text>
 				</xsl:when>
 			</xsl:choose>
 		</xsl:variable>
 
+		<xsl:variable as="xs:string?" name="skip">
```

Comment:
I assume you are intentionally reusing the variable name on lines 76 and 100. Just mentioning it in case your preferred style is to make variable names unique and the reuse was inadvertent.

Reply:
Yes, the variable is reused here. Probably in order to simplify the diff.